services/autoscaling/groups.go

```diff
@@ -0,0 +1,544 @@
+		value := int32(value.(int))
+		replica.Cores = &value
+	}
+
+	if value, ok := d.GetOk("replica_configuration.0.cpu_family"); ok {
+		value := autoscaling.CpuFamily(value.(string))
+		replica.CpuFamily = &value
+	} else {
+		replica.CpuFamily = nil
+	}
+
+	replica.Nics = GetNicsData(d)
+	if *replica.Nics == nil {
+		*replica.Nics = make([]autoscaling.ReplicaNic, 0)
+	}
+
+	if value, ok := d.GetOk("replica_configuration.0.ram"); ok {
+		value := int32(value.(int))
+		replica.Ram = &value
+	}
+
+	if volumes, err := GetVolumesData(d); err != nil {
+		return nil, err
+	} else {
+		replica.Volumes = volumes
+	}
+
+	return &replica, nil
+}
+
+func GetNicsData(d *schema.ResourceData) *[]autoscaling.ReplicaNic {
+	var nics []autoscaling.ReplicaNic
+
+	if value, ok := d.GetOk("replica_configuration.0.nic"); ok {
+		nicsSet := value.(*schema.Set)
+		if nicsSet != nil {
+			for _, val := range nicsSet.List() {
+				nicsMap := val.(map[string]any)
+				var nicEntry = autoscaling.NewReplicaNic(int32(nicsMap["lan"].(int)), nicsMap["name"].(string))
+				nicEntry.Dhcp = new(bool)
+				*nicEntry.Dhcp = nicsMap["dhcp"].(bool)
+				nics = append(nics, *nicEntry)
+			}
+		}
+	}
+
+	return &nics
+}
+
+func GetVolumesData(d *schema.ResourceData) (*[]autoscaling.ReplicaVolumePost, error) {
+	var volumes []autoscaling.ReplicaVolumePost
+
+	if value, ok := d.GetOk("replica_configuration.0.volume"); ok {
+		volumesSet := value.(*schema.Set)
+		if volumesSet != nil {
+			volumesValue := volumesSet.List()
+			if volumesValue != nil {
+				for index := range volumesValue {
+					volumeMap := volumesValue[index].(map[string]any)
+					var volumeEntry = *autoscaling.NewReplicaVolumePostWithDefaults()
+					if val, ok := volumeMap["name"]; ok {
+						volumeEntry.Name = new(string)
+						*volumeEntry.Name = val.(string)
+					}
+					if val, ok := volumeMap["image_alias"]; ok {
+						volumeEntry.ImageAlias = new(string)
+						*volumeEntry.ImageAlias = val.(string)
+					}
+					if val, ok := volumeMap["image"]; ok {
+						volumeEntry.Image = new(string)
+						*volumeEntry.Image = val.(string)
+					}
+					if val, ok := volumeMap["size"]; ok {
+						volumeEntry.Size = new(int32)
+						*volumeEntry.Size = int32(val.(int))
+					}
+					if val, ok := volumeMap["type"]; ok {
+						volumeEntry.Type = new(autoscaling.VolumeHwType)
+						*volumeEntry.Type = autoscaling.VolumeHwType(val.(string))
+					}
+					if *volumeEntry.Image == "" && *volumeEntry.ImageAlias == "" {
+						return nil, fmt.Errorf("it is mandatory to provide either public image or imageAlias that has cloud-init compatibility in conjunction with backup unit id property")
+					} else {
+						if val, ok := volumeMap["user_data"]; ok {
+							volumeEntry.UserData = new(string)
+							*volumeEntry.UserData = val.(string)
+						}
+					}
+					var publicKeys []string
+					sshKeys := volumeMap["ssh_keys"].([]any)
+					if len(sshKeys) != 0 {
```

Comment:
the length check is not really necessary since all logic is inside the `for` which won't run if there's no entries

Reply:
@cbeti-ionos solved in c9df22e0066ca69a7809de59cd124fe97fec9580